generatederrors.go

```diff
@@ -0,0 +1,2374 @@
+// This source file is part of the EdgeDB open source project.
+//
+// Copyright 2020-present EdgeDB Inc. and the EdgeDB authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file is auto generated. Do not edit!
+
+package edgedb
+
+import "fmt"
+
+// InternalServerError is an error.
+type InternalServerError interface {
+	Error
+	isInternalServerError()
```

Comment:
I assume we inject these methods to make interfaces unique from the standpoint of the Go compiler?

Reply:
Yes. This was the simplest way I could think of to implement the EdgeDB error hierarchy with interfaces and `errors.As()`. It doesn't require any custom `As()` implementation.